Source/carthage/Environment.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  Environment.swift
+//  Carthage
+//
+//  Created by J.D. Healy on 2/6/15.
+//  Copyright (c) 2015 Carthage. All rights reserved.
+//
+
+import CarthageKit
+import Foundation
+import LlamaKit
+
+internal func getEnvironmentVariable(variable: String) -> Result<String> {
+	let environment = NSProcessInfo.processInfo().environment
+
+	if let value = environment[variable] as String? {
+		return success(value)
+	} else {
+		return failure(CarthageError.MissingEnvironmentVariable(variable: variable).error)
+	}
+}
+
+/// Information about the possible parent terminal.
+internal struct Terminal {
```

Comment:
This would be easier to test if it weren’t (effectively) a singleton. I.e. if its properties weren’t `static`, and if `ColorArgument` were parameterized (or at least parameterizable) by it.

I’m not sure how much we care about testing around this, though. @jspahrsummers, any thoughts?